external/llvm-project/mlir/lib/Conversion/GPUCommon/GPUOpsLowering.cpp

```diff
@@ -181,6 +186,30 @@ GPUFuncOpLowering::matchAndRewrite(gpu::GPUFuncOp gpuFuncOp, OpAdaptor adaptor,
     }
   }
 
+  // Get memref type from function arguments and set the noalias to
+  // pointer arguments.
+  for (const auto &en : llvm::enumerate(gpuFuncOp.getArgumentTypes())) {
+    auto memrefTy = en.value().dyn_cast<MemRefType>();
+    NamedAttrList argAttr = argAttrs
+                                ? argAttrs[en.index()].cast<DictionaryAttr>()
+                                : NamedAttrList();
+    if (argAttr.empty())
+      continue;
+
+    if (memrefTy && argAttr.erase(LLVM::LLVMDialect::getNoAliasAttrName())) {
+      auto remapping = signatureConversion.getInputMapping(en.index());
+      for (size_t i = 0; i < remapping->size; i++)
+        if (llvmFuncOp.getArgument(remapping->inputNo + i)
+                .getType()
+                .template isa<LLVM::LLVMPointerType>()) {
+          llvmFuncOp.setArgAttr(remapping->inputNo + i,
```

Comment:
Correctness issue:
If `noalias` is present and the calling convention is not bare pointers (or, equivalently, if the remapping is of size > 1, currently), do **not** copy `noalias` and instead emit a warning. This is because memrefs, as you've probably noticed, have the allocation pointer and the aligned pointer, and those do, in fact, alias.

(In the future, we might want to add add an "aligned pointers only" scheme, but that's far future work we don't need)